crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -603,24 +421,29 @@ unsafe impl WorldQuery for EntityRef<'_> {
     }
 }
 
-/// SAFETY: Access is read-only.
-unsafe impl ReadOnlyWorldQuery for EntityRef<'_> {}
+/// SAFETY: `Self` is the same as `Self::ReadOnly`
+unsafe impl<'a> WorldQueryData for EntityRef<'a> {
+    type ReadOnly = Self;
+
+    fn shrink<'wlong: 'wshort, 'wshort>(item: Self::Item<'wlong>) -> Self::Item<'wshort> {
+        item
+    }
+}
+
+/// SAFETY:
+/// `fetch` accesses all components in a mutable way.
+/// This is sound because `update_component_access` and `update_archetype_component_access` set write access for all components and panic when appropriate.
```

Comment:
Is this comment correct? We shouldn't be setting write access for all components for a read-only view of entities.

Reply:
You are completely right. Have replaced with "SAFETY: access is read only"